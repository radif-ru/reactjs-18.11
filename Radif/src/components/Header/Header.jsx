import './Header.scss'

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types'

export const chatType =  {
  chats: PropTypes.array,
  // id: PropTypes.number,
  // messages: PropTypes.array,
  // name: PropTypes.string.isRequired
};

export class Header extends PureComponent {
  static propTypes = chatType
  render() {
    const { chats } = this.props;

    return (
      <header className='header'>
        <span className='header-name'>Список чатов:</span>
        {
          chats && chats.length && <div className='header-elements'>
            {chats.map((chat, idx) => <span key={idx}>{chat.name}; </span>)}
          </div>
        }
      </header>
    );
  }
}

// import './Header.scss'
//
// import React, {Component} from 'react';
// import PropTypes from 'prop-types'
//
// export const chatType =  {
//   id: PropTypes.number,
//   messages: PropTypes.array,
//   name: PropTypes.string.isRequired
// };
//
// export class Header extends Component {
//   static propTypes = chatType
//
//   static defaultProps = {
//     name: 'Header'
// }
//
//   render() {
//     const {name} = this.props;  // Теперь данный элемент не передаётся
//
//     return (
//       <div className='header'>
//         {name}
//       </div>
//     );
//   }
// }