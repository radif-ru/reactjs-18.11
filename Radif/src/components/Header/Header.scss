.header {
  display: flex;
  width: 100%;
  justify-content: center;
  background-color: moccasin;
  padding: 15px;
  font-size: 20px;
  color: green;
  &-name {
    padding-right: 5px;
  }
  &-elements {
    color: salmon;
  }
}