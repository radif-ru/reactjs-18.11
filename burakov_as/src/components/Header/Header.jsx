import React, { PureComponent } from 'react';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';

export class Header extends PureComponent{
  render() {
    const { chats } = this.props;

    return (
      <div>
        <Grid container>
          <Grid item xs={12}>
            <Paper >
              <header>
                {
                  chats && chats.length && <ul>
                    {chats.map((chat, idx) => <li key={idx}>{chat.name}</li>)}
                  </ul>
                }
              </header>
            </Paper>   
          </Grid>
        </Grid>
      </div>
    )
  }
}