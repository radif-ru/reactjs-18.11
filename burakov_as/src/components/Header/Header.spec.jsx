import React from 'react';
import { Header } from './Header';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('Header should be rendered', () => {
  const tree = renderer
  .create (<Header chats = {[{name:'Chat 1'}, {name: 'Chat 2'}]}/>)
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Header should be rendered as ul', () => {
  const header = shallow(<Header chats = {[{name:'Chat 1'}, {name: 'Chat 2'}]}/>);
  const lis = header.find('li');

  expect(lis.length).toBe(2);
  expect(lis.first().text()).toBe('Chat 1');
  expect(lis.last().text()).toBe('Chat 2');
  
});