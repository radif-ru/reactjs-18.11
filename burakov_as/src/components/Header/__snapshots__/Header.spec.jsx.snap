// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be rendered 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <header>
          <ul>
            <li>
              Chat 1
            </li>
            <li>
              Chat 2
            </li>
          </ul>
        </header>
      </div>
    </div>
  </div>
</div>
`;
